test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -386,25 +389,39 @@ object TestAspect extends TimeoutVariants {
    * An aspect that repeats the test a default number of times, ensuring it is
    * stable ("non-flaky"). Stops at the first failure.
    */
-  val nonFlaky: TestAspectPoly =
+  val nonFlaky
+    : TestAspect[Nothing, TestClock with TestConsole with TestRandom with TestSystem, Nothing, Any, Nothing, Any] =
     nonFlaky(100)
 
   /**
    * An aspect that repeats the test a specified number of times, ensuring it
    * is stable ("non-flaky"). Stops at the first failure.
    */
-  def nonFlaky(n: Int): TestAspectPoly =
-    new TestAspect.PerTest[Nothing, Any, Nothing, Any, Nothing, Any] {
-      def perTest[R >: Nothing <: Any, E >: Nothing <: Any, S >: Nothing <: Any](
-        test: ZIO[R, TestFailure[E], TestSuccess[S]]
-      ): ZIO[R, TestFailure[E], TestSuccess[S]] = {
-        def repeat(n: Int): ZIO[R, TestFailure[E], TestSuccess[S]] =
-          if (n <= 1) test
-          else test.flatMap(_ => repeat(n - 1))
-
-        repeat(n)
+  def nonFlaky(
+    n: Int
+  ): TestAspect[Nothing, TestClock with TestConsole with TestRandom with TestSystem, Nothing, Any, Nothing, Any] =
+    restoreTestEnvironment >>>
+      new TestAspect.PerTest[
+        Nothing,
+        TestClock with TestConsole with TestRandom with TestSystem,
```

Comment:
You don't need the type constraints here. This test aspect is just repeating a test, it doesn't have to know anything about the test environment. It may also make this read a little better to pull it out into a separate definition.

```scala
def nonFlaky(
    n: Int
  ): TestAspect[Nothing, TestClock with TestConsole with TestRandom with TestSystem, Nothing, Any, Nothing, Any] = {
  val nonFlaky = ???
  restoreTestEnvironment >>> nonFlaky
}
```

Same thing could probably help with the other cases where we have to define these inner test aspects.

Reply:
To do this I had to change restoreTestEnvironment to be lazy val due to the ordering (warning about referencing an uninitialized val) but I don't think this should be an issue

Reply:
Yes, I remember running into that too when I was playing with it. No problem. I would just change to `def` since they are not computationally expensive and will probably only be invoked once if at all in a typical application.